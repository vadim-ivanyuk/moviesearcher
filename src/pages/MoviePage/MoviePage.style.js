import styled from "styled-components";

export const MoviePageWrapper = styled.div`
  position: relative;
  width: 100%;
  height: auto;
`;

export const MoviePageContainer = styled.div`
  position: relative;
  width: 100%;
  max-width: 1140px;
  margin: 0 auto;
  padding: 0px 20px;
`;
