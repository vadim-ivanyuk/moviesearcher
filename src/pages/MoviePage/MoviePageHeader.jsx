import React from "react";
import PropTypes from "prop-types";
import { API_IMG_URL, API_IMG_URL_W1920 } from "../../utils/apies";
import { convertReleaseDate } from "../../utils/helpers";
import defaultBackdropImage from "../../img/not-found-backdrop.jpg";
import defaultPosterImage from "../../img/not-found-poster.jpg";
import {
  MovieHeader,
  MovieHeaderBackground,
  MovieHeaderBackgroundImage,
  MovieHeaderSingleBlock,
  MovieHeaderContainer,
  MovieHeaderPoster,
  MovieHeaderPosterImage,
  MovieHeaderDescription,
  MovieHeaderTitle,
  MovieHeaderTagLine,
  MovieHeaderVoteAvarageContainer,
  MovieHeaderVoteAvarage,
  MovieHeaderVoteAvarageText,
  MovieHeaderParagraph,
  MovieHeaderBoldTest,
  MovieHeaderOverviewTitle,
  MovieHeaderOverview,
} from "./MoviePageHeader.style";

const getValueFromArray = (arr = []) => {
  let resultArr = [];

  arr.forEach((item) => {
    resultArr.push(item.name);
  });

  return resultArr;
};

const getBackgropImage = (movie) => {
  if (!movie.backdrop_path) {
    return defaultBackdropImage;
  }

  const path = movie.belongs_to_collection
    ? movie.belongs_to_collection.backdrop_path
    : movie.backdrop_path;

  return `${API_IMG_URL_W1920}${path}`;
};

export const MoviePageHeader = (props) => {
  const { movie } = props;

  return (
    <MovieHeader>
      <MovieHeaderBackground>
        <MovieHeaderBackgroundImage src={getBackgropImage(movie)} alt="" />
      </MovieHeaderBackground>
      <MovieHeaderSingleBlock>
        <MovieHeaderContainer>
          <MovieHeaderPoster>
            <MovieHeaderPosterImage
              src={
                movie.poster_path
                  ? `${API_IMG_URL}${movie.poster_path}`
                  : defaultPosterImage
              }
              alt=""
            />
          </MovieHeaderPoster>
          <MovieHeaderDescription>
            <MovieHeaderTitle>{movie.title}</MovieHeaderTitle>
            {movie.tagline && (
              <MovieHeaderTagLine>{movie.tagline}</MovieHeaderTagLine>
            )}
            <MovieHeaderVoteAvarageContainer>
              <MovieHeaderVoteAvarage vote_avarage={movie.vote_average}>
                {String(movie.vote_average).length === 1
                  ? `${movie.vote_average}.0`
                  : movie.vote_average}
              </MovieHeaderVoteAvarage>
              <MovieHeaderVoteAvarageText>
                Пользовательский <br /> счёт
              </MovieHeaderVoteAvarageText>
            </MovieHeaderVoteAvarageContainer>
            <MovieHeaderParagraph>
              <MovieHeaderBoldTest>Статус</MovieHeaderBoldTest> - {movie.status}
            </MovieHeaderParagraph>
            <MovieHeaderParagraph>
              <MovieHeaderBoldTest>Дата выхода</MovieHeaderBoldTest> -{" "}
              {convertReleaseDate(movie.release_date).substr(0, 15)}
            </MovieHeaderParagraph>
            <MovieHeaderParagraph>
              <MovieHeaderBoldTest>Жанры</MovieHeaderBoldTest> -{" "}
              {getValueFromArray(movie.genres).join(", ")}
            </MovieHeaderParagraph>
            <MovieHeaderParagraph>
              <MovieHeaderBoldTest>Язык оригинала</MovieHeaderBoldTest> -{" "}
              {movie.original_language}
            </MovieHeaderParagraph>
            <MovieHeaderParagraph>
              <MovieHeaderBoldTest>Компания</MovieHeaderBoldTest> -{" "}
              {getValueFromArray(movie.production_companies).join(", ")}
            </MovieHeaderParagraph>
            <MovieHeaderParagraph>
              <MovieHeaderBoldTest>Страна</MovieHeaderBoldTest> -{" "}
              {getValueFromArray(movie.production_countries).join(", ")}
            </MovieHeaderParagraph>
            <MovieHeaderOverviewTitle>Обзор</MovieHeaderOverviewTitle>
            <MovieHeaderOverview tagline={!!movie.tagline}>
              {movie.overview}
            </MovieHeaderOverview>
          </MovieHeaderDescription>
        </MovieHeaderContainer>
      </MovieHeaderSingleBlock>
    </MovieHeader>
  );
};

MoviePageHeader.defaultProps = {
  movie: {},
};

MoviePageHeader.propTypes = {
  movie: PropTypes.object.isRequired,
};
