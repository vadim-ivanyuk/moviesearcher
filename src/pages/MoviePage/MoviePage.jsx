import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import queryString from "query-string";
import axios from "axios";
import { API_URL, API_KEY_MOVIE_DB_V3 } from "../../utils/apies";
import { MoviePageHeader } from "./MoviePageHeader";
import { MoviePageTrailers } from "./MoviePageTrailers";

const getMovieParams = (params) => ({
  language: params.language,
  api_key: API_KEY_MOVIE_DB_V3,
});

export const MoviePage = (props) => {
  const [movie, setMovie] = useState({});
  const filters = useSelector((store) => store.filters);

  const { id } = props.match.params;

  useEffect(() => {
    axios
      .get(
        `${API_URL}/movie/${id}?${queryString.stringify(
          getMovieParams(filters)
        )}`
      )
      .then(({ data }) => {
        setMovie(data);
      })
      .catch((error) => {
        console.log(error);
      });
  }, []);

  return (
    <div>
      <MoviePageHeader movie={movie} />
      <MoviePageTrailers movie_id={movie.id} />
    </div>
  );
};
