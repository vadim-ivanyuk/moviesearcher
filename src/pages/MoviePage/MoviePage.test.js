import React from "react";
import { configure, shallow } from "enzyme";
import EnzymeToJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { MoviePage } from "./MoviePage";

configure({ adapter: new Adapter() });

jest.mock("react-redux", () => ({
  useSelector: () => jest.fn(),
  useDispatch: () => jest.fn(),
}));

jest.mock("redux-form", () => ({
  formReducer: () => jest.fn(),
}));

describe("MoviePage", () => {
  const mockProps = {
    match: {
      params: 587807,
    },
  };

  it("first render snapshot", () => {
    const moviePage = shallow(<MoviePage {...mockProps} />);

    expect(EnzymeToJSON(moviePage)).toMatchSnapshot();
  });
});
