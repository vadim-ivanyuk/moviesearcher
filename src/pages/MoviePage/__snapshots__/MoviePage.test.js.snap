// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage first render snapshot 1`] = `
<div>
  <MoviePageHeader
    movie={Object {}}
  />
  <MoviePageTrailers />
</div>
`;
